compiler/sigmatch.nim

```diff
@@ -1425,6 +1425,8 @@ proc typeRelImpl(c: var TCandidate, f, aOrig: PType,
     considerPreviousT:
       if a.kind == tyGenericInst and a.sons[0] == f:
         bindingRet isGeneric
+      if a == f:
+        bindingRet isEqual
```

Comment:
Generic matches can never be ``isEqual``. This needs to be ``isGeneric`` but even then it's not clear it's the proper way to fix this bug.